changes to execute slim sim in native python for development

MLeon96 opened the pull request:
add dotenv to read .env, renamed folder to run, modified docker-compose

both, main.py and docker-compose are using the .env in /run to load env added a few lines in main.py to check for keywords of the env or load it

added if in main.py to check a few env variables for execution in docker (env already loaded) or execution in native python (variable not known, load .env).

Commit:
changes to execute slim sim in native python for dev
add dotenv to read .env, renamed folder to run, modified docker-compose

both, main.py and docker-compose are using the .env in /run to load env
added a few lines in main.py to check for keywords of the env or load it

check a few env variables for execution in docker (env already loaded)
or execution in native python (variable not known, load .env).

ndunkelivi commented on src/main.py:
```diff
@@ -1,4 +1,14 @@
+if 'UUID' and 'NAME' in os.environ:  # check if key exists which means that environment is already loaded by docker
+    pass
+else:  # otherwise load environment for native python execution
+    dotenv = Path('../run/.env')
```
check if local .env file (and certificate file and geometry.json) exists in src folder (instead of example folder), and load environment from there
add comprehensive error message and suggestions for fix, if .env does not exist
add instructions to README (e.g. manually copy .env to src folder)

MLeon96 replied:
done and simplified as discussed

ndunkelivi commented on README.md:
````diff
@@ -9,10 +9,13 @@ Run with the default settings:
 docker run helyosframework/agent_helyos_slim_simulator
 ```
 
````
keep 'example' folder name

ndunkelivi commented on run/docker-compose.yml:
```diff
@@ -0,0 +1,62 @@
+            - ./custom-geometry.json:/app/customization/geometry.json       # To modify reported geometry.
+            - ./ca_certificate.pem:/app/ca_certificate.pem                  # To verify rbmq server url.
+        environment:
+            # view .env file for variable list, defaults and comments
```
Add comment 
#The contents of the .env file are automatically loaded into the container
#For configuration options of the agent simulator see the .env file

remove reassignment of env variables below

MLeon96 replied:
will not work, done as discussed.

Commit:
edit Readme, reduced dotenv loading,
changed checkin behvaiour for better errormessages
added some comments and troubleshooting

ndunkelivi committed:
Update README.md

ndunkelivi committed:
use helyos_agent_sdk version 0.8.1

ndunkelivi committed:
update documentation

- remove redundant comments from .env file and docker-compose.yml
- add example to mount src code into docker container via docker-compose.yml
- revise instructions for native and docker deployment in README.md

ndunkelivi committed:
change default agent name; add comments

ndunkelivi committed:
outsource checkin routine to dedicated function

ndunkelivi committed:
Update Readme.md

ndunkelivi committed:
update Readme.md

cviolbarbosa commented:
@ndunkelivi please build an image locally and test it against the helyOS core integration tests before releasing. (You may copy these tests here in the future). After this, I think this branch is ready to be merged.

ndunkelivi commented:
> @ndunkelivi please build an image locally and test it against the helyOS core integration tests before releasing. 

Thank you, I did so
